marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/job/finishJob.sjs

```diff
@@ -18,8 +18,46 @@
 xdmp.securityAssert("http://marklogic.com/data-hub/privileges/run-step", "execute");
 
 const Job = require("/data-hub/5/flow/job.sjs");
+const dhProv = require('/data-hub/5/provenance/dh-provenance.xqy');
+const config = require("/com.marklogic.hub/config.sjs");
 
 var jobId;
 var jobStatus;
 
-Job.getRequiredJob(jobId).finishJob(jobStatus, fn.currentDateTime()).update();
+function updateJobProvenance(jobId, jobEndTime) {
+  const options = {
+    "endTime": jobEndTime
+  }
+  const provCollectionQuery = cts.collectionQuery("http://marklogic.com/provenance-services/record");
+  const provIdAttributeQuery = cts.elementAttributeValueQuery(fn.QName("http://www.w3.org/ns/prov#", "activity"),
+    fn.QName("http://www.w3.org/ns/prov#", "id"), fn.concat("job:", jobId));
+  const jobProvQuery = cts.andQuery([provCollectionQuery, provIdAttributeQuery]);
+
+  const stagingJobProvRecordUri = fn.documentUri(fn.head(
+    xdmp.invokeFunction(function () {
+      return cts.search(jobProvQuery);
+    }, {
+      database: xdmp.database(config.STAGINGDATABASE)
+    })
+  ));
+
+  let finalJobProvRecordUri = fn.documentUri(fn.head(
+    xdmp.invokeFunction(function () {
+      return cts.search(jobProvQuery);
+    }, {
+      database: xdmp.database(config.FINALDATABASE)
+    })
+  ));
+
+  if(stagingJobProvRecordUri && fn.docAvailable(stagingJobProvRecordUri)) {
```

Comment:
The fn.docAvailable call isn't necessary, and actually won't work for one of the two. 

Also, since you only want the URI back, use cts.uris instead of cts.search - it's faster to get the URI from the URI lexicon as opposed to loading the document from disk.

Reply:
It works for both when a flow with steps having staging and final databases as target databases.

Reply:
Also, changed cts.search to cts.uris

Reply:
But finishJob is called against the jobs database. The fn.docAvailable call will then be made against the jobs database. But these two URIs were just fetched from the staging and final databases. So I'm confused - I would think fn.docAvailable will always return false, unless the URIs are also in the jobs database??